### DISCUSSION TITLE
Quorum queues backup

### DISCUSSION THREAD

**Discussion by Alexanator** [rabbitmq/rabbitmq-server#4158] [Questions]
Hi! Got question about backuping messages from quorum queues.
Official documentation [says](https://www.rabbitmq.com/backup.html#manual-messages-backup) that in case of using mirrored queues It needs to stop entire cluster before backup.
But what about quorum queues? Can't find any information about it. Do I need stop entire cluster or is it enough to stop and start nodes one by one?
Please help if you have any experience.

**Comment by mkuratczyk** [comment]
In most cases, backing up messages isn't really needed - most likely (hopefully) they will be consumed before your backup finishes. And if the cluster is running, you keep getting messages during the backup so most recent messages won't be backed up.
Can you explain what you want to achieve?

**Reply by Alexanator** [reply]
Hello and thanks for answer.
I need to backup messages in json format which stores in quorum queues. This messages are criticaly important ande we do not want to lose any in case of accident.

**Comment by michaelklishin** [comment]
At various points we have considered "backup-like" solutions, e.g. non-voting replicating Raft member support in Ra https://github.com/rabbitmq/ra/pull/96 (this would be an extension to Raft, the protocol itself does not have such concept).
They were dropped because the complexity or nuances outweighed the benefits.

Tanzu RabbitMQ has a warm standby feature that allows you to keep a back up cluster around with a certain configurable backlog
of messages that can be restored in case of disaster recovery. This is the most appropriate approach in our team's opinion as
backing up highly transient data controlled by applications is not exactly the same as backing up a directory of static files.
